src/couch/src/couch_file.erl

```diff
@@ -264,6 +265,19 @@ rename_file(Original) ->
         Else -> Else
     end.
 
+rename_dir(RootDelDir, Original, DbName) ->
+    DbDir = binary_to_list(DbName) ++ "_design",
+    [DbPureName | _R] = lists:reverse(filename:split(binary_to_list(DbName))),
```

Comment:
The discussion was reflected in summary of this commit (https://github.com/apache/couchdb/pull/597/commits/9b1dbfd5a2d770a4f2c2a5cb79c6e4575dad36c1).
```
 - Introduce dbname directory between .recovery and shards so that
   all view files belonging to the same db can be located at the
   the central place. Thus, these view files can be easily sorted
   by atime at /srv/data/.recovery level.
   When these view files from same db needs to be recovered, they
   can be easily moved back to view directory.
```

However, considering the user information in DBName and situation where "/" is included in dbname, I changed back to `/srv/view_index/.recovery/shards/e0000000-ffffffff/testdb1.1500350972_design` approach.